commands/index.js

```diff
@@ -16,31 +16,57 @@ function generateRandomSessionId() {
+function getCurrencySymbol(currency) {
+  return config.CURRENCY_SYMBOLS[currency] ? config.CURRENCY_SYMBOLS[currency] : '';
+}
+
+function isObject(key) {
+  return key && typeof key === 'object';
+}
+
+function runNLPResponse(api_ai, bot, message) {
+  const request = api_ai.textRequest(message.text, {
     sessionId: generateRandomSessionId()
   });
 
   request.on('response', response => {
-    switch (response.result.metadata.intentName) {
+    if(!response.result.actionIncomplete) {
+      switch (response.result.metadata.intentName) {
 
-    case config.NLP_INTENTS.WORD_DEFINITION:
-      definitionCommand(bot, message, response.result.parameters['any']);
-      break;
+      case config.NLP_INTENTS.WORD_DEFINITION:
+        definitionCommand(bot, message, response.result.parameters['any']);
+        break;
 
-    case config.NLP_INTENTS.GHS_CONVERSION: // eslint-disable-line no-case-declarations
-      const amount = response.result.parameters['input-money']; // the captured amount
-      const currency = response.result.parameters['input-currency'];
-      const currencyValue = config.CURRENCIES[currency] !== undefined ? config.CURRENCIES[currency] : currency;
-      const currencySymbol = config.CURRENCY_SYMBOLS[currency] !== undefined ? config.CURRENCY_SYMBOLS[currency] : "";
+      case config.NLP_INTENTS.GHS_CONVERSION: // eslint-disable-line no-case-declarations
+        if(isObject(response.result.parameters['unit-currency'])) {
+          const amount = response.result.parameters['unit-currency'].amount; // the captured amount
+          const currency = response.result.parameters['unit-currency'].currency;
+          forexConversionCommand(currency, getCurrencySymbol(currency), amount)(bot, message);
+        } else {
+          bot.reply(message, 'Sorry, your query should be in the form `Convert 50 usd`, with the 3 character currency code.');  
+        }
         
-      forexConversionCommand(currencyValue, currencySymbol, amount)(bot, message);
-      break;
+        break;
+
+      case config.NLP_INTENTS.CONVERT_FROM_X_TO_Y: // eslint-disable-line no-case-declarations
+        if(isObject(response.result.parameters['unit-currency'])) {
+          const inputAmount = response.result.parameters['unit-currency'].amount; // the captured amount
+          const inputCurrency = response.result.parameters['unit-currency'].currency;
+          const resultCurrency = response.result.parameters['currency-name'];
+          forexConversionCommand(inputCurrency, getCurrencySymbol(inputCurrency), inputAmount, resultCurrency)(bot, message);
+        } else {
+          bot.reply(message, 'Sorry, your query should be in the form `Convert 50 usd to ngn`, with the 3 character currency codes.');  
+        }
+        break;
 
-    default:
-      break;
+      default:
+        break;
+      }
+    } else {
+      bot.reply(message, 'Sorry, your query seems incomplete. Please refer to the docs for usage rules');     
```

Comment:
A link to docs would be useful in here, maybe save it as a config so we can set it during app initialising (so a ENV).